src/platform/intel/cavs/lib/pm_runtime.c

```diff
@@ -62,6 +64,25 @@ static inline void cavs_pm_runtime_force_host_dma_l1_exit(void)
 	spin_unlock_irq(_prd->lock, flags);
 }
 
+static inline void cavs_pm_runtime_request_dsp_d0(uint32_t req)
+{
+	uint32_t flags;
+	struct cavs_pm_runtime_data *pprd = _prd->platform_data;
+
+	irq_local_disable(flags);
+
```

Comment:
I also think it's probably worth a comment here describing why we dont use any atomic ops here (iirc, there was some coherence reason betweeen memory cores ?)

Reply:
Commented.